File: regexlib/python_re2_test_file/regexlib_3748.py
# 3748
# ^(?:[\w]+[\&amp;\-_\.]*)+@(?:(?:[\w]+[\-_\.]*)\.(?:[a-zA-Z]{2,}?))$
# EXPONENT
# nums:5
# EXPONENT AttackString:""+"__"*16+"!1 __EOA(i or ii)"

import re2 as re
from time import perf_counter

regex = """^(?:[\w]+[\&amp;\-_\.]*)+@(?:(?:[\w]+[\-_\.]*)\.(?:[a-zA-Z]{2,}?))$"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "" + "__" * i * 1 + "!1 __EOA(i or ii)"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *1}: took {DURATION} seconds!")

File: regexlib/2021-5-15/javascript_test_file/regexlib_3748.js
/* 3748
 * ^(?:[\w]+[\&amp;\-_\.]*)+@(?:(?:[\w]+[\-_\.]*)\.(?:[a-zA-Z]{2,}?))$
 * EXPONENT
 * nums:3
 * EXPONENT AttackString:""+"__"*16+"!1 __EOA(i or ii)"
 */
var REGEX = /^(?:[\w]+[\&amp;\-_\.]*)+@(?:(?:[\w]+[\-_\.]*)\.(?:[a-zA-Z]{2,}?))$/
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = '' + '__'.repeat(i*1) + '!1 __EOA(i or ii)'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 1 + ": " + time_cost+" ms")
}